### PR TITLE
KeIgaras' fix PR.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Opening by request, alternative to other busted PR.
Original PR: #2387

### CODE REVIEW

**Review by comma — DISMISSED** [review]

File: code/modules/atmospherics/components/unary/vent_scrubber.dm [hunk 1]
```diff
@@ -77,8 +77,10 @@
 		icon_state = "weld"
 	else if((stat & NOPOWER) || !use_power)
 		icon_state = "off"
+	else if(scrubbing == "exchange")
```
Comment (afterthought2) [inline code/modules/atmospherics/components/unary/vent_scrubber.dm]: While it's true that the current code is nonsensical, there do seem to be defines you should be using for this.

File: code/game/objects/items/weapons/tools/weldingtool.dm [hunk 2]
```diff
@@ -109,11 +109,12 @@
 	if(!status && istype(W, /obj/item/stack/material/rods))
 		var/obj/item/stack/material/rods/R = W
 		R.use(1)
-		add_fingerprint(user)
 		user.drop_from_inventory(src)
 		var/obj/item/flamethrower/F = new(get_turf(src), src)
+		F.add_fingerprint(user)
```
Comment (afterthought2) [inline code/game/objects/items/weapons/tools/weldingtool.dm]: It's highly unlikely to be correct to manually add fingerprints like this.

**Review by afterthought2 — COMMENTED** [review]
I find the blood data manipulation concerning: is this really the way to do it?

Other comments are obviously minor.

File: code/game/objects/items/weapons/tools/weldingtool.dm (hunk 2 above)
Comment (keIgaras) [reply code/game/objects/items/weapons/tools/weldingtool.dm:114]: now prints are also being added everywhere, for an alternative option it needs to be refactored

Comment (kelgaras) [reply code/game/objects/items/weapons/tools/weldingtool.dm]: there is no choice, they are added so everywhere

File: code/modules/atmospherics/components/unary/vent_scrubber.dm (hunk 1 above)
Comment (kelgaras) [reply code/modules/atmospherics/components/unary/vent_scrubber.dm]: you mean, i need to swap ``"exchange"`` for ``SCRUBBER_EXCHANGE`` define?

File: code/game/objects/items/weapons/tools/weldingtool.dm (hunk 2 above)
Comment (afterthought2) [reply code/game/objects/items/weapons/tools/weldingtool.dm]: I'm not actually sure this one is replicated in equip code, but it likely should be.

Comment (kelgaras) [reply code/game/objects/items/weapons/tools/weldingtool.dm]: looked at process ``put_in_hands``, ``drop_from_inventory``, and ``remove_from_mob`` and didn't found where fingerprints has be added

Comment (MistakeNot4892) [reply code/game/objects/items/weapons/tools/weldingtool.dm]: Added an add_fingerprint() call to base item equipped().

File: code/modules/atmospherics/components/unary/vent_scrubber.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/atmospherics/components/unary/vent_scrubber.dm]: Done.

**Comment by MistakeNot4892** [comment]
Reviewed the blood data manipulation and I agree - the main issue appeared to be the original code returning early rather than trying to retrieve data if it wasn't provided, rather than the blood data being null. `reset_blood()` should be populating the mob's DNA hashes etc in their blood data already.

File: code/modules/atmospherics/components/unary/vent_scrubber.dm (hunk 1 above)
Comment (afterthought2) [reply code/modules/atmospherics/components/unary/vent_scrubber.dm]: If you did, you didn't push the commit.

Comment (MistakeNot4892) [reply code/modules/atmospherics/components/unary/vent_scrubber.dm]: Shit

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**